### PR TITLE
Add test coverage for Date.prototype.toTemporalInstant

### PR DESCRIPTION
**Pull request by anba**

### CODE REVIEW

File: test/built-ins/Date/prototype/toTemporalInstant/not-a-constructor.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+assert.sameValue(
+  isConstructor(Date.prototype.toTemporalInstant),
+  false,
+  'isConstructor(Date.prototype.toTemporalInstant) must return false'
+);
+
+assert.throws(TypeError, function() {
+  var date = new Date(0);
+  new date.toTemporalInstant();
```
Comment (Ms2ger) [inline test/built-ins/Date/prototype/toTemporalInstant/not-a-constructor.js:26]:
```suggestion
var date = new Date(0);
assert.throws(TypeError, function() {
  new date.toTemporalInstant();
```

File: test/built-ins/Date/prototype/toTemporalInstant/this-value-non-date.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+// Copyright (C) 2024 André Bargull. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-date.prototype.totemporalinstant
+description: >
+  Behaviour when "this" value is an Object without a [[DateValue]] internal slot
```
Comment (Ms2ger) [inline test/built-ins/Date/prototype/toTemporalInstant/this-value-non-date.js:7]: Please also test `Date.prototype` itself and a function

File: test/built-ins/Date/prototype/toTemporalInstant/this-value-non-object.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
+// Copyright (C) 2024 André Bargull. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-date.prototype.totemporalinstant
+description: >
+  Behaviour when "this" value is not an Object
```
Comment (Ms2ger) [inline test/built-ins/Date/prototype/toTemporalInstant/this-value-non-object.js:7]: Missing BigInt

File: test/built-ins/Date/prototype/toTemporalInstant/this-value-valid-date.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+assert.sameValue(
+  new Date(123_456_789).toTemporalInstant().epochNanoseconds,
+  123_456_789_000_000n,
+  "date after start of the epoch"
+);
+
+assert.sameValue(
+  new Date(-123_456_789).toTemporalInstant().epochNanoseconds,
+  -123_456_789_000_000n,
+  "date before start of the epoch"
```
Comment (Ms2ger) [inline test/built-ins/Date/prototype/toTemporalInstant/this-value-valid-date.js:33]: Nit: the epoch is a point in time, so "start of the epoch" should be "the (Unix) epoch" (three times)

**Review by Ms2ger — APPROVED** [review]
Thanks!

File: test/built-ins/Date/prototype/toTemporalInstant/this-value-non-object.js (hunk 3 above)
Comment (anba) [reply test/built-ins/Date/prototype/toTemporalInstant/this-value-non-object.js:7]: Filed #4315

File: test/built-ins/Date/prototype/toTemporalInstant/not-a-constructor.js (hunk 1 above)
Comment (anba) [reply test/built-ins/Date/prototype/toTemporalInstant/not-a-constructor.js:26]: Filed #4316

**Comment by anba** [comment]
Addressed all review comments.

**Pull request merged**